File: binding/src/libqtestsupport_gui.cpp
#include <QEventPoint>
#include <QPoint>
#define WORKAROUND_INNER_CLASS_DEFINITION_QTest__QTouchEventSequence
#include <QWindow>
#include <qtestsupport_gui.h>
#include "libqtestsupport_gui.h"
#include "libqtestsupport_gui.hxx"

QTest__QTouchEventSequence* QTest__QTouchEventSequence_new(const QTest__QTouchEventSequence* param1) {
	 return new VirtualQTestQTouchEventSequence(*param1);
}

QTest__QTouchEventSequence* QTest__QTouchEventSequence_Press(QTest__QTouchEventSequence* self, int touchId, const QPoint* pt) {
	return new QTest::QTouchEventSequence(self->press(touchId, *pt));
}

QTest__QTouchEventSequence* QTest__QTouchEventSequence_Move(QTest__QTouchEventSequence* self, int touchId, const QPoint* pt) {
	return new QTest::QTouchEventSequence(self->move(touchId, *pt));
}

QTest__QTouchEventSequence* QTest__QTouchEventSequence_Release(QTest__QTouchEventSequence* self, int touchId, const QPoint* pt) {
	return new QTest::QTouchEventSequence(self->release(touchId, *pt));
}

QTest__QTouchEventSequence* QTest__QTouchEventSequence_Stationary(QTest__QTouchEventSequence* self, int touchId) {
	return new QTest::QTouchEventSequence(self->stationary(touchId));
}

bool QTest__QTouchEventSequence_Commit(QTest__QTouchEventSequence* self, bool processEvents) {
	return self->commit(processEvents);
}

QTest__QTouchEventSequence* QTest__QTouchEventSequence_Press3(QTest__QTouchEventSequence* self, int touchId, const QPoint* pt, QWindow* window) {
	return new QTest::QTouchEventSequence(self->press(touchId, *pt, window));
}

QTest__QTouchEventSequence* QTest__QTouchEventSequence_Move3(QTest__QTouchEventSequence* self, int touchId, const QPoint* pt, QWindow* window) {
	return new QTest::QTouchEventSequence(self->move(touchId, *pt, window));
}

QTest__QTouchEventSequence* QTest__QTouchEventSequence_Release3(QTest__QTouchEventSequence* self, int touchId, const QPoint* pt, QWindow* window) {
	return new QTest::QTouchEventSequence(self->release(touchId, *pt, window));
}

// Base class handler implementation
QTest__QTouchEventSequence* QTest__QTouchEventSequence_QBaseStationary(QTest__QTouchEventSequence* self, int touchId) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_Stationary_IsBase(true);
	return new QTest::QTouchEventSequence(vqtest__qtoucheventsequence->stationary(touchId));
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventSequence_OnStationary(QTest__QTouchEventSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_Stationary_Callback(reinterpret_cast<VirtualQTestQTouchEventSequence::QTest__QTouchEventSequence_Stationary_Callback>(slot));
}
}

// Base class handler implementation
bool QTest__QTouchEventSequence_QBaseCommit(QTest__QTouchEventSequence* self, bool processEvents) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_Commit_IsBase(true);
	return vqtest__qtoucheventsequence->commit(processEvents);
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventSequence_OnCommit(QTest__QTouchEventSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_Commit_Callback(reinterpret_cast<VirtualQTestQTouchEventSequence::QTest__QTouchEventSequence_Commit_Callback>(slot));
}
}

// Derived class handler implementation
QPoint* QTest__QTouchEventSequence_MapToScreen(QTest__QTouchEventSequence* self, QWindow* window, const QPoint* pt) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
	return new QPoint(vqtest__qtoucheventsequence->mapToScreen(window, *pt));
	} else {
	return new QPoint(self->QTest::QTouchEventSequence::mapToScreen(window, *pt));
}
}

// Base class handler implementation
QPoint* QTest__QTouchEventSequence_QBaseMapToScreen(QTest__QTouchEventSequence* self, QWindow* window, const QPoint* pt) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_MapToScreen_IsBase(true);
	return new QPoint(vqtest__qtoucheventsequence->mapToScreen(window, *pt));
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventSequence_OnMapToScreen(QTest__QTouchEventSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_MapToScreen_Callback(reinterpret_cast<VirtualQTestQTouchEventSequence::QTest__QTouchEventSequence_MapToScreen_Callback>(slot));
}
}

// Derived class handler implementation
QEventPoint* QTest__QTouchEventSequence_Point(QTest__QTouchEventSequence* self, int touchId) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
	return new QEventPoint(vqtest__qtoucheventsequence->point(touchId));
	} else {
	return new QEventPoint(self->QTest::QTouchEventSequence::point(touchId));
}
}

// Base class handler implementation
QEventPoint* QTest__QTouchEventSequence_QBasePoint(QTest__QTouchEventSequence* self, int touchId) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_Point_IsBase(true);
	return new QEventPoint(vqtest__qtoucheventsequence->point(touchId));
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventSequence_OnPoint(QTest__QTouchEventSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_Point_Callback(reinterpret_cast<VirtualQTestQTouchEventSequence::QTest__QTouchEventSequence_Point_Callback>(slot));
}
}

// Derived class handler implementation
QEventPoint* QTest__QTouchEventSequence_PointOrPreviousPoint(QTest__QTouchEventSequence* self, int touchId) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
	return new QEventPoint(vqtest__qtoucheventsequence->pointOrPreviousPoint(touchId));
	} else {
	return new QEventPoint(self->QTest::QTouchEventSequence::pointOrPreviousPoint(touchId));
}
}

// Base class handler implementation
QEventPoint* QTest__QTouchEventSequence_QBasePointOrPreviousPoint(QTest__QTouchEventSequence* self, int touchId) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_PointOrPreviousPoint_IsBase(true);
	return new QEventPoint(vqtest__qtoucheventsequence->pointOrPreviousPoint(touchId));
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventSequence_OnPointOrPreviousPoint(QTest__QTouchEventSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_PointOrPreviousPoint_Callback(reinterpret_cast<VirtualQTestQTouchEventSequence::QTest__QTouchEventSequence_PointOrPreviousPoint_Callback>(slot));
}
}

// Derived class handler implementation
void QTest__QTouchEventSequence_OperatorAssign(QTest__QTouchEventSequence* self, const QTest__QTouchEventSequence* param1) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
	vqtest__qtoucheventsequence->operator=(*param1);
	} else {
	self->QTest::QTouchEventSequence::operator=(*param1);
}
}

// Base class handler implementation
void QTest__QTouchEventSequence_QBaseOperatorAssign(QTest__QTouchEventSequence* self, const QTest__QTouchEventSequence* param1) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_OperatorAssign_IsBase(true);
	vqtest__qtoucheventsequence->operator=(*param1);
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventSequence_OnOperatorAssign(QTest__QTouchEventSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventsequence = dynamic_cast<VirtualQTestQTouchEventSequence*>(self);
	if (vqtest__qtoucheventsequence && vqtest__qtoucheventsequence->isVirtualQTestQTouchEventSequence) {
vqtest__qtoucheventsequence->setQTest__QTouchEventSequence_OperatorAssign_Callback(reinterpret_cast<VirtualQTestQTouchEventSequence::QTest__QTouchEventSequence_OperatorAssign_Callback>(slot));
}
}

void QTest__QTouchEventSequence_Delete(QTest__QTouchEventSequence* self) {
    delete self;
}

File: binding/src/libqtestsupport_gui.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qtest-qtoucheventsequence.html
pub const qtest__qtoucheventsequence = struct {

    /// New constructs a new QTest::QTouchEventSequence object.
    pub fn New(param1: ?*anyopaque) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_new(@ptrCast(param1));
    }


    pub fn Press(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_Press(@ptrCast(self), touchId, @ptrCast(pt));
    }

    pub fn Move(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_Move(@ptrCast(self), touchId, @ptrCast(pt));
    }

    pub fn Release(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_Release(@ptrCast(self), touchId, @ptrCast(pt));
    }

    pub fn Stationary(self: ?*anyopaque, touchId: i32) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_Stationary(@ptrCast(self), touchId);
    }

    /// Allows for overriding the related default method
    pub fn OnStationary(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) QtC.QTest__QTouchEventSequence) void {
        qtc.QTest__QTouchEventSequence_OnStationary(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseStationary(self: ?*anyopaque, touchId: i32) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_QBaseStationary(@ptrCast(self), touchId);
    }

    pub fn Commit(self: ?*anyopaque, processEvents: bool) bool {
        return qtc.QTest__QTouchEventSequence_Commit(@ptrCast(self), processEvents);
    }

    /// Allows for overriding the related default method
    pub fn OnCommit(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) bool) void {
        qtc.QTest__QTouchEventSequence_OnCommit(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCommit(self: ?*anyopaque, processEvents: bool) bool {
        return qtc.QTest__QTouchEventSequence_QBaseCommit(@ptrCast(self), processEvents);
    }

    pub fn MapToScreen(self: ?*anyopaque, window: ?*anyopaque, pt: ?*anyopaque) QtC.QPoint {
        return qtc.QTest__QTouchEventSequence_MapToScreen(@ptrCast(self), @ptrCast(window), @ptrCast(pt));
    }

    /// Allows for overriding the related default method
    pub fn OnMapToScreen(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) QtC.QPoint) void {
        qtc.QTest__QTouchEventSequence_OnMapToScreen(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMapToScreen(self: ?*anyopaque, window: ?*anyopaque, pt: ?*anyopaque) QtC.QPoint {
        return qtc.QTest__QTouchEventSequence_QBaseMapToScreen(@ptrCast(self), @ptrCast(window), @ptrCast(pt));
    }

    pub fn Point(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventSequence_Point(@ptrCast(self), touchId);
    }

    /// Allows for overriding the related default method
    pub fn OnPoint(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) QtC.QEventPoint) void {
        qtc.QTest__QTouchEventSequence_OnPoint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePoint(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventSequence_QBasePoint(@ptrCast(self), touchId);
    }

    pub fn PointOrPreviousPoint(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventSequence_PointOrPreviousPoint(@ptrCast(self), touchId);
    }

    /// Allows for overriding the related default method
    pub fn OnPointOrPreviousPoint(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) QtC.QEventPoint) void {
        qtc.QTest__QTouchEventSequence_OnPointOrPreviousPoint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePointOrPreviousPoint(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventSequence_QBasePointOrPreviousPoint(@ptrCast(self), touchId);
    }

    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QTest__QTouchEventSequence_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnOperatorAssign(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QTest__QTouchEventSequence_OnOperatorAssign(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseOperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QTest__QTouchEventSequence_QBaseOperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    pub fn Press3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, window: ?*anyopaque) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_Press3(@ptrCast(self), touchId, @ptrCast(pt), @ptrCast(window));
    }

    pub fn Move3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, window: ?*anyopaque) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_Move3(@ptrCast(self), touchId, @ptrCast(pt), @ptrCast(window));
    }

    pub fn Release3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, window: ?*anyopaque) QtC.QTest__QTouchEventSequence {
        return qtc.QTest__QTouchEventSequence_Release3(@ptrCast(self), touchId, @ptrCast(pt), @ptrCast(window));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QTest__QTouchEventSequence_Delete(@ptrCast(self));
    }
};
